src/candid.ts

```diff
@@ -0,0 +1,163 @@
+  }
+  return einBaseField.id;
+};
+
+const isValidEin = (s: string): boolean => (
+  /^\d{2}-?\d{7}$/.test(s)
+);
+
+const getEinsFromPdc = async (baseUrl: string, token: AccessTokenSet) => {
+  const einBaseFieldId = await getEinBaseFieldId(baseUrl, token);
+  const proposals = await getProposals(baseUrl, token);
+
+  const eins = new Set(proposals.entries
+    .flatMap((proposal) => proposal.versions[0]?.fieldValues)
+    .filter(<T>(x: T | undefined): x is T => typeof x !== 'undefined')
+    .filter(({ applicationFormField }) => applicationFormField.baseFieldId === einBaseFieldId)
+    .map(({ value }) => value)
+    .filter(isValidEin));
+  return [...eins];
+};
+
+const candid: CommandModule = {
+  command: 'candid',
+  describe: 'Interact with the Candid Premier API',
+  builder: (y) => (y
+    .option('candid-api-key', {
+      describe: 'Candid Premier API key; get from account management at https://dashboard.candid.org/',
+      demandOption: true,
+      global: true,
+      type: 'string',
+    })
+    .command(
+      'lookup <ein>',
+      'Fetch and display information about an organization by its EIN',
+      {
+        'output-file': {
+          alias: 'write',
+          describe: 'Write organization information to the specified JSON file',
+          normalize: true,
+          type: 'string',
+        },
+      },
+      async (args) => {
+        const result = await getCandidProfile(
+          args.candidApiKey as string,
+          args.ein as string,
+        );
+
+        if (typeof args.outputFile === 'string') {
+          await writeFile(
+            args.outputFile,
+            JSON.stringify(result, null, 2),
+          );
+          logger.info(`Wrote Candid data for ${args.ein as string} to ${args.outputFile}`);
+        } else {
+          logger.info('Candid result', { result });
+        }
+      },
+    )
+    .command(
+      'update <ein>',
+      'Fetch information about an organization by its EIN, and upload to the PDC',
+      oidcOptions,
+      async (args) => {
+        const token = await getToken(
+          args.oidcBaseUrl as string,
+          args.oidcClientId as string,
+          args.oidcClientSecret as string,
+        );
+        const data = await getCandidProfile(args.candidApiKey as string, args.ein as string);
+        await postPlatformProviderData(
+          args.pdcApiBaseUrl as string,
+          token,
+          args.ein as string,
+          'Candid',
+          data,
+        );
+        logger.info(`Wrote data for ${args.ein as string} to PDC`);
+      },
+    )
+    .command(
+      'update-all',
+      'Update Candid profiles for all PDC proposals',
+      oidcOptions,
+      async (args) => {
+        const token = await getToken(
+          args.oidcBaseUrl as string,
+          args.oidcClientId as string,
+          args.oidcClientSecret as string,
+        );
+        const eins = await getEinsFromPdc(args.pdcApiBaseUrl as string, token);
```

Comment:
I'm also not a fan of this pattern of passing a token + the base URL around. An alternative I considered is to make a PDC client class that initializes an Axios client, keeps it in the instance, and has a bunch of methods that use it; another alternative is to pass that `AxiosInstance` around directly. I'm kinda leaning towards the latter, but haven't implemented it yet.